docs/self-managed/zeebe-deployment/operations/disk-space.md

```diff
@@ -8,11 +8,12 @@ Zeebe uses the local disk for storage of its persistent data. Therefore, if the
 
 To prevent the system from reaching an unrecoverable state, Zeebe expects a minimum size of free disk space available. If this limit is violated, the broker rejects new requests to allow the operations team to free more disk space, and allows the broker to continue to update its state.
 
-Zeebe can be configured with the following settings for the disk usage watermarks:
+Zeebe can be configured with the following settings for the disk usage:
 
-- **zeebe.broker.data.diskUsageMonitoringEnabled**: Configure if disk usage should be monitored (default: true)
-- **zeebe.broker.data.diskUsageReplicationWatermark**: The fraction of used disk space before the replication is paused (default: 0.99)
-- **zeebe.broker.data.diskUsageCommandWatermark**: The fraction of used disk space before new user commands are rejected (default: 0.97), this must be less than `diskUsageReplicationWatermark`.
-- **zeebe.broker.data.diskUsageMonitoringInterval**: The interval in which the disk space usage is checked (default 1 second)
-
-For **production** use cases, we recommend setting the values for `diskUsageReplicationWatermark` and `diskUsageCommandWatermark` to smaller values, for example `diskUsageReplicationWatermark=0.9` and `diskUsageCommandWatermark=0.8`.
+- **zeebe.broker.data.disk.enablemonitoring**: Configure if disk usage should be monitored (default: true)
+- **zeebe.broker.data.disk.monitoringInterval**: The interval in which the disk space usage is checked (default: 1 second)
+- **zeebe.broker.data.disk.freeSpace.replication**: When the free space available is less than this value, Zeebe pauses receiving replicated events. (default: 1GB)
+  - For **production** use cases, we recommend to increase this value and set it approximately to `number of partitions x logSegmentSize + 1GB`.
```

Comment:
@npepinpe Please check if the recommendation makes sense.

Reply:
Wasn't this based on expected snapshot size as well? As recovery from this state is contingent on receiving a snapshot/compacting your log. So I would want to have enough free space to receive a snapshot. 

The recommendation now is we keep at least space for 1 segment, and if we don't have space for one segment, stop replicating. Same for processing/writes, but two segments.

I think that's fair, as we don't have a real size for snapshots.

Reply:
> The recommendation now is we keep at least space for 1 segment, and if we don't have space for one segment, stop replicating. Same for processing/writes, but two segments.

We recommend +1GB besides the segments I assumed this will be enough to take 1 or 2 new snapshots for all partitions together. We can also assume, all partitions don't receive a snapshot at the same time. So we don't have to leave one snapshotSize per partition. Of course, this is not enough for large snapshots, but they are a special case anyway.